src/main/java/org/cirdles/topsoil/data/NumberField.java

```diff
@@ -43,7 +44,13 @@ public String toString(Number number) {
 
             @Override
             public Number fromString(String string) {
-                return Double.valueOf(string);
+                
+               //Check if the string parameter is a number
+                if (string.matches("\\d+(\\.\\d+)?")) {
```

Comment:
This regex is way too long, I used a try-catch instead, it is as efficient. For the tests, we can't build them on the CI server.